lxd/backup/config/backup_config.go

```diff
@@ -11,13 +11,29 @@ const DefaultMetadataVersion = api.BackupMetadataVersion1
 // MaxMetadataVersion represents the latest supported metadata version.
 const MaxMetadataVersion = api.BackupMetadataVersion2
 
+// Volume represents the config of a volume including its snapshots.
+type Volume struct {
+	// Make sure to have the embedded structs fields inline to avoid nesting.
+	api.StorageVolume `yaml:",inline"`
+
+	Snapshots []*api.StorageVolumeSnapshot `yaml:"snapshots,omitempty"`
+}
+
 // Config represents the config of a backup that can be stored in a backup.yaml file (or embedded in index.yaml).
 type Config struct {
-	Container       *api.Instance                `yaml:"container,omitempty"` // Used by VM backups too.
-	Snapshots       []*api.InstanceSnapshot      `yaml:"snapshots,omitempty"`
-	Pool            *api.StoragePool             `yaml:"pool,omitempty"`
-	Profiles        []*api.Profile               `yaml:"profiles,omitempty"`
-	Volume          *api.StorageVolume           `yaml:"volume,omitempty"`
+	Version   uint32                  `yaml:"version,omitempty"`
+	Instance  *api.Instance           `yaml:"instance,omitempty"`
+	Snapshots []*api.InstanceSnapshot `yaml:"snapshots,omitempty"`
+	Pools     []*api.StoragePool      `yaml:"pools,omitempty"`
```

Comment:
As I was looking through the code it occurred to me that whilst `Snapshots` makes sense as a slice as order is important, the `Pools` and `Volumes` fields do not have order. Would it make sense to make these maps, keyed on their respective pool or volume name? This would make loading a specific pool/volume for read/update easier.

Reply:
oh now, ofc, we cant do that for volumes as they may have the same name in different pools. Ignore me.